### PR TITLE
Better validation and tests for notificationRecipients

### PR DESCRIPTION
**Pull request by adambraimbridge**
If you set 'slack.users' or 'slack.channels' in 'flag.notificationRecipients', they must be arrays.

cc @GlynnPhillips (If you wouldn't mind sanity-checking, please)

image

### CODE REVIEW

File: server/lib/flagValidator.js [hunk 1]
```diff
@@ -16,14 +16,51 @@ validations.set('A Flag must have a description', flag => {
 });
 
 validations.set('A Flag must have a `notificationRecipients` (object) with a `slack.users` (array), a `slack.channels` (array), or both.', flag => {
```
Comment (GlynnPhillips) [inline server/lib/flagValidator.js]: I would change this to only check if `notificationRecipients` exists and have the message be "A Flag must have a notificationRecipients object"

File: server/lib/flagValidator.js [hunk 2]
```diff
@@ -16,14 +16,51 @@ validations.set('A Flag must have a description', flag => {
...
+	try {
+		hasSlackChannels = flag.notificationRecipients.slack.channels instanceof Array && flag.notificationRecipients.slack.channels.length > 0;
+	}
+	catch(error) {
+		// flag.notificationRecipients.slack.channels was not defined — which is fine.
+	}
+	return (hasSlackUsers || hasSlackChannels);
+});
+
+validations.set('If you set `slack.users` or `slack.channels` in `flag.notificationRecipients`, they must be arrays.', flag => {
```
Comment (GlynnPhillips) [inline server/lib/flagValidator.js]: I would then make this check that the arrays exist and that they are arrays with a message something like

"The notificationRecipients object must contain a users and or channels array"

**Comment by GlynnPhillips** [comment]
Would it be worth having some validation on the values of channels and users? For instance initial confusion is caused by not knowing if you have to add the `#` and `@` might be overkill but just an idea.

**Comment by commuterjoy** [comment]
old - reopen and merge if needed please.

**Pull request closed without merging**